### ISSUE TITLE
Reputation System

### ISSUE THREAD

**Issue by lonnieezell** [lonnieezell/forum-example#188]
We should port the reputation system over from the current CI boards. I think it works well and I'd hate to lose that.

**Comment by michalsn** [comment]
I have never used the reputation system on our forum - it took me a while to find a button. 

It looks interesting, although I don't quite understand the point of awarding a neutral reputation (0 points). As I think about it, a negative reputation is also a bit strange. I'm not quite sure what should happen to give someone a negative reputation. Spam accounts don't count because they will be deleted - and we already have a thread/post reporting system.

Questions:

* How is it that some users can give +/- 1 point and others +/- 2 points?
* Should marking a post as a reply automatically give +2 points to reputation? If so, should removing a mark take away those points?

**Comment by lonnieezell** [comment]
When you click on the little hand button at the bottom of a post (horrible icon for this....) it brings up a modal where you can select the range from -2 to +2. You can then leave a comment with it. 

I don't like the idea of automatically giving reputation. I think it should be a manual action. 

It works as a "like" system, basically, where you can say you really don't like what they're saying (I've seen that in the past when people were being jerks on the forum), or you really like the comments, their actions, solution, etc. I don't know that neutral is all that useful. 

Writing this out, I could see this being a little more visual where the modal provides emoticons to represent those. If we went that route, though, then we'd almost have to do the Facebook style where it shows each emoticon and a count, so maybe that's more bother than its worth....

**Comment by michalsn** [comment]
How about trying to do it in an even simpler way?

As far as I can see 99% of people don't use the option to leave a comment when adding reputation points. If we drop the comments, we don't lose too much, and we get rid of another thing that potentially needs to be moderated.

Maybe we should just make "up" and "down" arrows next to each post (at the bottom of each post) - something like on Reddit?

This way we can increase the engagement of users in giving reputation. The downside of this will be that the table in the database will probably grow because of increased engagement.

Even with this super simple approach, we will be able to import the old reputation.

**Comment by michalsn** [comment]
Hmm... after thinking about it for a moment, what I'm proposing is probably a bit too complicated compared to the current reputation system.

**Comment by michalsn** [comment]
The "up" and "down" arrows would have to be related to the user, not the post...

**Comment by datamweb** [comment]
I disagree with the current forum reputation system in general. This system is really funny in my opinion.
You can imagine that Kenji answers almost all the questions, but he has scored 179 points so far. This system is not fair.
In my opinion, the system should automatically consider things to increase points automatically, in addition to that, users should be able to rate.
For example, if an answer is selected by the user, the score should be added automatically.

I didn't think about it carefully, but what is clear is that I don't think the current system of the forum is fair. In addition, this system should be implemented in a way that encourages users to participate and respond.

**Comment by lonnieezell** [comment]
I believe MyBB has what everyone is talking about split into two different parts: 

- the Repuration system, which is based solely on other user's view of you
- an automated system that is based on the number of posts and you can set their title in the admin to allow you to know how busy someone has been on the site. 

I think separating the two is valid, honestly. Just because you post a lot doesn't mean you're helpful. You could be  a troll leaving shit posts on everyone's threads. I don't think any of us would consider that user to have a great reputation. However, a person could be newer, with fewer posts than the troll, and be very helpful so they're getting a higher reputation from other users. 

As for the comments, I agree most people don't leave a comment. However, the ones with comments are the ones that actually make me feel good about it, if that makes sense. I'm not opposed to making a it a more simple "up" and "down" system, though. I almost never go look at my reputation stuff, and I don't know how much most people do, but I think having the system is definitely a good thing for rewarding people in a small way. If we simplify, do we need to provide the details publicly about what the reputation was built on? I don't think so. We could move that page to the Account section if someone is interested.

**Comment by michalsn** [comment]
Ok. While I believe that marking a post as an answer should automatically award reputation points, it would deprive us of the ability to leave a comment. So, keeping it separate makes sense - we can always revisit it later.

What we should do with the reputation points if the user who gave the reputation points is deleted or the post related to these points is deleted? Are we leaving the points or deleting them as well? I'm leaning towards removing them.

Does the reputation points can be also given to the user directly? MyBB allows this - not sure if we want to keep it that way.

**Comment by lonnieezell** [comment]
Was looking at some other forums and stumbled across [Discourse](https://www.discourse.org/features#), which actually looks really nice. One thing that really stood out to me was their [Trust Levels](https://blog.discourse.org/2018/06/understanding-discourse-trust-levels/), though. 

Basically, trust levels grant certain community features to users based on their activity. It tracks things like how long they've read, how many posts they've read, replied to, etc, and the more they do the things that builds a community the more trust they have on the site. At higher levels they actually get some features that moderators would typically have, allowing them to move threads to other categories, rename, mark posts as spam, flag, etc. So the community does a better job of policing itself. 

I really like this idea. That's one thing we've struggled a bit with on the CI forums as moderators have started falling off. If we're going to a new system, I think we have the chance to discard the stuff we don't like, and it sounds like the current reputation system is no one's favorite thing. 

What do you guys think about the Trust levels? Oh - and more badges users can earn based on various activities on the site? Basically it helps gamify being part of the community.

**Comment by michalsn** [comment]
I like this idea. Gamification and badges are also cool features.

We would have to determine which parts of the Trust System we would like to adopt.... all of them or just some of them to start with.

The only part of the Trust System that I don't like is collecting information about time spent reading posts... sounds like a privacy issue to me.

**Comment by lonnieezell** [comment]
I'm fine not including time reading. I understand why there were including it, but I don't know it will hurt things much to disregard that. 

I guess we need to start a new discussion with the features and levels and see what we like, what we don't, and if there's anything we want to add to it.
